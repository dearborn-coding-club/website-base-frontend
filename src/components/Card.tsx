import { CSSProperties, ReactNode } from "react"
import "../css/Card.css"

type propType = {
  imageStyle?: CSSProperties
  imageTextStyle?: CSSProperties
  textStyle?: CSSProperties
  imagetext?: ReactNode
  text?: ReactNode
}

export default function Card(prop: propType) {
  return (
    <div className="Card">
      <div className="card-image" style={prop.imageStyle ?? {}}>
        <span style={prop.imageTextStyle ?? {}}>{prop.imagetext}</span>
      </div>
      <div style={prop.textStyle ?? {}}>{prop.text}</div>
    </div>
  )
}
