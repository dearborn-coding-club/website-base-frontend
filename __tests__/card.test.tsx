import renderer from 'react-test-renderer';
import Card from '../src/components/Card'

it('Renders the Card component successfully', () => {
  const component = renderer.create(
    <Card />
  )
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})