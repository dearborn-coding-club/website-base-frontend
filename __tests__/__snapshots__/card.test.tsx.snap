// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Card component successfully 1`] = `
<div
  className="Card"
>
  <div
    className="card-image"
    style={{}}
  >
    <span
      style={{}}
    />
  </div>
  <div
    style={{}}
  />
</div>
`;

exports[`Renders the component class 1`] = `
<div
  className="Card"
>
  <div
    className="card-image"
    style={{}}
  >
    <span
      style={{}}
    />
  </div>
  <div
    style={{}}
  />
</div>
`;
